### PR TITLE
[11.0][ADD] hr_department_channel

### PR DESCRIPTION
**Pull request by jarroyomorales**
This module allows you to easily create private channels for departments. It will automatically add and remove users from the channel when they join or leave a department.

To create a new channel for a department go to it's form and click the `Create Channel` button. Then you just need to indicate the name of the channel.

@etobella

### CODE REVIEW

File: hr_department_channel/models/hr_department.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+class HrDepartment(models.Model):
+    _inherit = 'hr.department'
+
+    department_channel_ids = fields.One2many(
+        'mail.channel', readonly=True, string='Channels',
+        inverse_name='department_id',
+    )
+
+    department_channel_count = fields.Integer(
+        compute="_compute_department_channel_count", store=True,
```
Comment (etobella) [inline hr_department_channel/models/hr_department.py]: why do you store it?

File: hr_department_channel/wizards/wizard_create_department_channel.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+            members |= department_id.manager_id.user_id
+        partners = members.mapped('partner_id')
+        department_id.department_channel_ids = [(0, 0, {
+            'name': self.name,
+            'public': 'private',
+            'channel_partner_ids': [
+                (6, 0, partners.ids)
+            ]
+        })]
+        return {'type': 'ir.actions.act_window_close'}
```
Comment (etobella) [inline hr_department_channel/wizards/wizard_create_department_channel.py]: I think this is not necessary

**Review by etobella — APPROVED** [review]
Just a few comments

**Comment by jarroyomorales** [comment]
@etobella Comments attended

**Pull request closed without merging**